lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -292,22 +295,26 @@ def compile(node, options)
     config
   end
 
-  # Turn our host name into a node object.
-  def find_node(name, environment, transaction_uuid, configured_environment)
+  # Use indirection to find the node associated with a given request
+  def find_node(facts, request)
+    name = request.key || request.node
+    environment = request.environment
+    transaction_uuid = request.options[:transaction_uuid]
+    configured_environment = request.options[:configured_environment]
+
     Puppet::Util::Profiler.profile(_("Found node information"), [:compiler, :find_node]) do
       node = nil
       begin
         node = Puppet::Node.indirection.find(name, :environment => environment,
                                              :transaction_uuid => transaction_uuid,
-                                             :configured_environment => configured_environment)
+                                             :configured_environment => configured_environment,
+                                             :facts => facts)
```

Comment:
If the `facts` from the request is nil (from https://github.com/puppetlabs/puppet/pull/5866/files#diff-a499fd5869d90e9967707b6fb6f445f3R45), we end up passing a `:facts => nil` param above. Is that ok? Should we instead omit that hash entry?